backends/ebpf/ebpfTable.cpp

```diff
@@ -872,4 +872,169 @@ void EBPFCounterTable::emitTypes(CodeBuilder* builder) {
     builder->endOfStatement(true);
 }
 
+////////////////////////////////////////////////////////////////
+
+EBPFValueSet::EBPFValueSet(const EBPFProgram* program, const IR::P4ValueSet* p4vs,
+                           cstring instanceName, CodeGenInspector* codeGen)
+        : EBPFTableBase(program, instanceName, codeGen), size(0), pvs(p4vs) {
+    CHECK_NULL(pvs);
+    valueTypeName = "u32";  // map value is not used, so its type can be anything
+
+    // validate size
+    if (pvs->size->is<IR::Constant>()) {
+        auto sc = pvs->size->to<IR::Constant>();
+        if (sc->fitsUint())
+            size = sc->asUnsigned();
+        if (size == 0)
+            ::error(ErrorType::ERR_OVERLIMIT,
+                    "Size must be a positive value less than 2^32, got %1% entries", pvs->size);
+    } else {
+        ::error(ErrorType::ERR_UNSUPPORTED_ON_TARGET,
+                "Size of value_set must be know at compilation time: %1%", pvs->size);
+    }
+
+    // validate type
+    if (pvs->elementType->is<IR::Type_Bits>() || pvs->elementType->is<IR::Type_Tuple>()) {
+        // no restrictions
+    } else if (pvs->elementType->is<IR::Type_Name>()) {
```

Comment:
In general you can do `typeMap->getType(pvs->elementType, true)`.
This should cover all cases, including nested typedefs, etc.

Reply:
`getType` returns instance of `IR::Type_Type`. This is intended behavior?

Reply:
Use getTypeType